hydra-explorer/src/Hydra/Explorer.hs

```diff
@@ -73,18 +93,25 @@ logMiddleware tracer app' req sendResponse = do
         }
   app' req sendResponse
 
-httpApp :: Tracer IO APIServerLog -> GetHeads -> Application
-httpApp tracer getHeads =
-  logMiddleware tracer $ serve api $ server getHeads
+httpApp :: Tracer IO APIServerLog -> GetHeads -> GetTick -> Application
+httpApp tracer getHeads getTick =
+  logMiddleware tracer $ serve api $ server getHeads getTick
 
-observerHandler :: TVar IO ExplorerState -> [HeadObservation] -> IO ()
+observerHandler :: TVar IO ExplorerState -> [ChainObservation] -> IO ()
 observerHandler explorerState observations = do
   atomically $
     modifyTVar' explorerState $
       aggregateHeadObservations observations
 
 readModelGetHeadIds :: TVar IO ExplorerState -> GetHeads
-readModelGetHeadIds = readTVarIO
+readModelGetHeadIds explorerStateTVar = do
+  ExplorerState{heads} <- readTVarIO explorerStateTVar
+  pure heads
+
+readModelGetTick :: TVar IO ExplorerState -> GetTick
+readModelGetTick explorerStateTVar = do
+  ExplorerState{tick} <- readTVarIO explorerStateTVar
+  pure tick
```

Comment:
Could: DRY this into a single function and provide that to the `httpApp`.

For example, a `getState :: IO ExplorerState` function and then the handler implementations would be:

```
getHeads = getState <&> \ExplorerState{heads} -> heads

getTick = getState <&> \ExplorerState{tick} -> tick
```